### DISCUSSION TITLE
Odd bug when referencing shared workflow by tag

### DISCUSSION THREAD

**Discussion by chrisgilbert** [community/community#62403] [Actions]
### Select Topic Area

Bug

### Body

I have been testing and changing reusable workflows for the past week, and have now switched most of our deploys to using workflows from a central private repo. It's mostly been fine, but I had a weird bug appear this afternoon that I wanted to report.

I've been making changes to the v3 tag several times throughout the day. When rerunning consuming workflows, it's been working fine. Indeed, I ran the same workflow with the same tag in the PR right before merging, and it worked fine.

I then tried changing the tag to point at v3.1.1 instead, and it deployed successfully. Pointing it back to v3 then failed again. It seems like maybe something is cached somewhere - the SHA doesn't seem right mentioned in the error message. It's also complaining about different dependent workflow files each time - and all are present for the tag version on the source repo, as shown for my screenshot.

<img width="1051" alt="Screenshot 2023-08-01 at 17 38 39" src="https://github.com/community/community/assets/361112/6048ea05-8868-4f01-8eb5-ca5d61070f40">
<img width="1189" alt="Screenshot 2023-08-01 at 17 38 57" src="https://github.com/community/community/assets/361112/a6a75c4e-043c-4ed3-97d4-f077c402625d">
<img width="739" alt="Screenshot 2023-08-01 at 17 45 28" src="https://github.com/community/community/assets/361112/ba0f60de-a0a8-497c-895a-1b2f7051c2cc">
<img width="696" alt="Screenshot 2023-08-01 at 17 48 44" src="https://github.com/community/community/assets/361112/c0eee4a9-ad96-4395-a29d-83ec471da81c">

This hasn't happened in any other repos - I am going to try recreating the tag/waiting and see if either make any difference.

**Comment by ricardo-almeida-mck** [comment]
I think I’m stumbling upon the same issue. Any luck solving it ?
